### PR TITLE
add proposal of robot account enhancement

### PR DESCRIPTION
**Pull request by wy65701436**

### CODE REVIEW

File: proposals/new/Robot-Account-2.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+Robot Account is a System Administrator and Project Administrator operation in Harbor.
+
+* User Stories
+
+1.  As a system admin, I can create a system level robot account with the selected projects and access scope.
+2.  As a system admin, I can edit a system level robot account to enhance/reduce the access scope.
+3.  As a system admin, I can edit a system level robot account to enhance/reduce the project scope.
+4.  As a system admin, I can extend the expiration data of a system level robot account.
+5.  As a system admin, I can view the token of a system level robot account and refresh the token.
```
Comment (reasonerjt) [inline proposals/new/Robot-Account-2.md:40]: I recall there's requirement to `edit` the token, for example I can set the password of a robot account to `passw0rd`

File: proposals/new/Robot-Account-2.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+## Personas and User Stories
+
+This section lists the user stories for the different personas interacting with robot account.
+
+* Personas
+
+Robot Account is a System Administrator and Project Administrator operation in Harbor.
+
+* User Stories
+
```
Comment (reasonerjt) [inline proposals/new/Robot-Account-2.md:35]: There's also requirement not forcing the user to use `robot$xxxx` as username?

File: proposals/new/Robot-Account-2.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+ project_id int,
+ /*
+  token string used as the password of robot account.
+  For v2.2, it stores the secret.
+  */
+ token varchar(255),
+ /*
+  permissions string used as the access scope.
+  */
+ permissions varchar(1024),
```
Comment (reasonerjt) [inline proposals/new/Robot-Account-2.md:77]: can we extract the scope to another table?  
In future we may associate such scopes with a customized role or access token that maybe shared by different Harbor instances.

Comment (reasonerjt) [inline proposals/new/Robot-Account-2.md:77]: What if the permission conflicts with the project id?

**Comment by ChristianCiach** [comment]
I just created the issue #13384. I am not sure why the robot user cannot currently be used to sync private projects, but I guess it has something to do with the `/v2/_catalog` endpoint that the robot user is not allowed to access. But this is just a guess.

File: proposals/new/Robot-Account-2.md [hunk 4]
```diff
@@ -0,0 +1,430 @@
+Proposal: Robot account enhancement
+
+Author: Yan Wang
+
+Discussion: [goharbor/harbor#10774](https://github.com/goharbor/harbor/issues/10774) [goharbor/harbor#11286](https://github.com/goharbor/harbor/issues/11286)
+
+## Abstract
+
+Robot account limited into one specific project, it cannot access multiple projects.
```
Comment (danielpacak) [inline proposals/new/Robot-Account-2.md:9]: Would it be possible to grant permissions to access registry catalog /v2/_catalog for such new robot account?

File: proposals/new/Robot-Account-2.md [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+  project_id 
+    0 means, it's a system level robot
+    non-0 means, it's a project level robot
+  */
+ project_id int,
+ /*
+  secret string used as the password of robot account.
+  For v2.2, it stores the secret.
+  */
+ secret varchar(2048),
```
Comment (reasonerjt) [inline proposals/new/Robot-Account-2.md:74]: Please make sure this is hash, not plain text.

File: proposals/new/Robot-Account-2.md [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+
+![mock_ui](../images/robot-account-2/robot_create.png)
+
+### To Be Discussed 
+
+* The pre-defined access scope for robot creation, no Spec so far.
+* (P1)Whether to provide Kubernetes pull secret for the robot account.
+* (P1)Whether to provide Docker credentials config for the robot account.
+
+## Security Concern
```
Comment (reasonerjt) [inline proposals/new/Robot-Account-2.md:423]: Please update the proposal to conclude how these concerns are addressed.

**Comment by MrD2** [comment]
Is this coming with v2.2.0

**Comment by Vad1mo** [comment]
> Is this coming with v2.2.0

@MrD2 join tomorrow the community meeting at 14:00 CEST I’ll being that topic up. As we need that as well

**Comment by christian-korneck** [comment]
> join tomorrow the community meeting at 14:00 CEST

JFYI: as to a message on the mailing list today's community meeting has been cancelled (next date 24-Feb-2021, 14:00 CEST ?)

**Review by heww — APPROVED** [review]

**Review by ninjadq — APPROVED** [review]

**Pull request merged**